dmutils/filters.py

```diff
@@ -40,3 +40,8 @@ def format_links(text):
         return formatted_text
     else:
         return text
+
+
+def nbsp(text):
+    """Replace spaces with nbsp."""
+    return Markup('&nbsp;'.join(text.split(' ')))
```

Comment:
We need to escape the original text before marking it as safe. This means we can't use this filter with anything that could contain HTML, but those parts should be `Markup` objects at this point anyway.